### PR TITLE
#167735052 User Profile information from travel request page 

### PR DESCRIPTION
**Pull request by amoskeyz**
**What does this PR do?**

  Gets User Profile information from travel request page 

  **Description of Task to be completed?**

 >- create remember profile endpoint
>- get user information on travel request if remember profile is true 

  **How should this be manually tested?**

>- Clone the repo and run npm install to download all packages, then use npm run test to test the endpoint
>- git clone this branch  <a href='https://github.com/andela/iceman-backend/profile-information-from-travel-request-page-167735052'>branch</a>
>- cd into the project
>- checkout to the branch using the command git checkout `profile-information-from-travel-request-page-167735052`
>- Create a Postgres Database
>- Create a `.env` file and set `PORT`, `JWT_SECRET` and `DB variables` as specified in the .env.example file.
>- Run the command `npm install` in the terminal.
>- Open a terminal and run npm `test-dev`
>- After the tests have passed. Run `npm run server` in the console.

  **Any background context you want to provide?**

  N/A

  **What are the relevant stories?**

  <a href = 'https://www.pivotaltracker.com/n/projects/2382051/stories/167735052'>#167735052</a>

  **Screenshots (if appropriate)**

  
image

  **Questions:**

 N/A

### CODE REVIEW

**Comment by Annmary12** [comment]
@amoskeyz , resolve conflict and get you teammates to review your PR. I can't see anyone approval.

**Comment by amoskeyz** [comment]
> @amoskeyz , resolve conflict and get you teammates to review your PR. I can't see anyone approval.

ok

File: src/services/authService.js [hunk 1]
```diff
@@ -255,4 +255,37 @@ export default class AuthService {
...
+
+    return 'Remember Profile Information Successfully Updated';
+  }
+
+  /**
+   * @param {object} req reqest object
+   * @return {string} - success message;
+   */
+  static async getRememberProfile(req) {
+    const { id } = req.user;
```
Comment (cleave3) [inline src/services/authService.js]: You can destructure req.user as an argument like this `static async getRememberProfile({ user: { id } })` that way your code is shorter and cleaner and it follows the team's convention. Please also effect this in other necessary places

File: src/services/authService.js [hunk 2]
```diff
@@ -255,4 +255,37 @@ export default class AuthService {
 
     return 'User Role Assigned Successfully';
   }
+
+  /**
+   * @param {object} req reqest object
+   * @return {string} - success message;
+   */
+  static async rememberUserProfile(req) {
```
Comment (cleave3) [inline src/services/authService.js]: You can apply restructuring here like this  `static rememberUserProfile( { body: { remember profile }, user: { id } })`

Comment (amoskeyz) [reply src/services/authService.js]: ok

Comment (Annmary12) [reply src/services/authService.js]: @cleave3, destructing there makes the function a bit busy, it is better to do that inside the function. cc: @amoskeyz

Comment (cleave3) [reply src/services/authService.js]: OK. Noted

File: src/controllers/authController.js [hunk 3]
```diff
@@ -151,4 +151,34 @@ export default class AuthController {
...
   }
+
+  /**
+   * @param {object} req request object
+   * @param {object} res response object
+   * @return {object} message
+   */
+  static async rememberProfile(req, res) {
+    try {
+      const isRemember = await AuthService.rememberUserProfile(req);
```
Comment (Annmary12) [inline src/controllers/authController.js]: Move this function `await AuthService.rememberUserProfile(req);` to create request function methods.
Check:
```
if(req.body.remember) {
await AuthService.rememberUserProfile(req)
}

```

File: src/controllers/authController.js [hunk 4]
```diff
@@ -151,4 +151,34 @@ export default class AuthController {
...
+      badRequest(res, error);
+    }
+  }
+
+  /**
+   * @param {object} req request object
+   * @param {object} res response object
+   * @return {object} message
+   */
+  static async getUserInformation(req, res) {
```
Comment (Annmary12) [inline src/controllers/authController.js]: Remove this function and reuse @elikeyz function

File: src/routes/auth.js [hunk 5]
```diff
@@ -8,8 +8,9 @@ import {
   passwordResetSchema,
   verifyEmail,
   LogInSchema,
-  profileSchema,
   roleSchema,
+  rememberSchema,
```
Comment (Annmary12) [inline src/routes/auth.js]: remove `rememberSchema`

File: src/validation/schemas.js [hunk 6]
```diff
@@ -103,3 +103,10 @@ export const requestSchema = Joi.object().keys({
   status: Joi.string(),
   accommodation: Joi.string().required().error(() => ({ message: 'Accommodation is required' }))
 });
+
+export const rememberSchema = Joi.object().keys({
```
Comment (Annmary12) [inline src/validation/schemas.js]: Move this schema to request schema

File: src/tests/role.test.js [hunk 7]
```diff
@@ -192,4 +192,77 @@ describe('Assign User Role', () => {
...
+
+  //   it('should not get user details with invalid token', async () => {
+  //     const res = await chai.request(app)
+  //       .get(`${URL_PREFIX}/user_information`);
+
+  //     res.should.have.status(401);
+  //     res.body.should.have.property('error');
+  //     res.body.error.should.equal('Access Denied, No token provided');
+  //   });
+  // });
```
Comment (Annmary12) [inline src/tests/role.test.js]: Why did you comment role test?

Comment (amoskeyz) [reply src/tests/role.test.js]: oops!!!  was suppose to remove that......  i have removed the comment.

**Pull request merged**